File: ServerSide/BallPassSchedulePattern/Players_QrcodeTypes/m_QRCodeMapInitializerPlayerDir/QRCodeMapInitializerPlayer.py
import os, sys
sys.path.append("../..")
from Players_CommonPlayers.SuperPlayerDir.SuperPlayer import SuperPlayer

class QRCodeMapInitializerPlayer(SuperPlayer):
    def __init__(self):
        super().__init__()  # スーパークラスの初期化メソッドを呼び出す
        self.version = 2  # QRコードのバージョン
        self.grid_size = 25  # QRコードのグリッドサイズ
        self.mode_charNumInfo_checksum_bitlist = None

    def return_my_name(self):
        return "QRCodeMapInitializerPlayer"

    def main(self):
        """
        このメソッド実行直前に、
        このスーパークラスのメンバ変数
        one_time_world_instanceに、
        最新のworldインスタンスを代入しています。
        
        このメソッド終了後、
        メインで使用しているworld_instanceを
        このスーパークラスのメンバ変数
        one_time_world_instanceで上書きし、
        更新しています。
        """
        
        """ 入力 """
        self.mode_charNumInfo_checksum_bitlist = self.one_time_world_instance.checksumPlayer.mode_charNumInfo_checksum_bitlist
        
        # QRコードマップを初期化して保持
        print(f"{self.return_my_name()}が実行されました。")
        
        """ 出力 """
        self.mode_charNumInfo_checksum_bitlist # データ
        self.version # QRコードのバージョン
        self.grid_size # QRコードのグリッドサイズ
  
        
        # 自身を更新
        self.one_time_world_instance.qRCodeMapInitializerPlayer = self

        return "Completed"

File: InstanceSide/BallPassSchedulePattern/Players_QRcodeTypes/m_QRCodeMapInitializerPlayerDir/QRCodeMapInitializerPlayer.cs
using UdonSharp;
using UnityEngine;

public class QRCodeMapInitializerPlayer : SuperPlayer
{
    // メンバ変数（Pythonのインスタンス変数に相当）
    public int Version = 2; // QRコードのバージョン
    public int GridSize = 25; // QRコードのグリッドサイズ
    public float[] ModeCharNumInfoChecksumBitlist; // モード/文字数情報/チェックサムのビットリスト

    // RinaNumpyをアタッチ
    public RinaNumpy rinaNumpy; // Unityエディタでアタッチすることを想定

    // 依存するプレイヤー
    public UdonSharpBehaviour ChecksumPlayer; // Unityでアタッチすることを想定

    // 自分の名前を返すメソッド
    public string ReturnMyName()
    {
        return "QRCodeMapInitializerPlayer";
    }

    public bool QRCodeMapInitializerPlayerReset()
    {
        return true;
    }

    // メイン処理
    public string ExecuteMain()
    {
        /*
         * UnityエディタでChecksumPlayerとRinaNumpyInstanceをアタッチすることを前提に処理を進める
         */

        // ChecksumPlayerからデータを取得
        QRCodeMapInitializerPlayer checksumPlayer = (QRCodeMapInitializerPlayer)ChecksumPlayer;
        ModeCharNumInfoChecksumBitlist = checksumPlayer.ModeCharNumInfoChecksumBitlist;

        // RinaNumpyを活用した処理例: 配列の正規化
        float mean = rinaNumpy.Mean_FloatArray(ModeCharNumInfoChecksumBitlist);
        float std = rinaNumpy.Std_FloatArray(ModeCharNumInfoChecksumBitlist);
        ModeCharNumInfoChecksumBitlist = rinaNumpy.Subtract_FloatArray_Float(ModeCharNumInfoChecksumBitlist, mean);
        ModeCharNumInfoChecksumBitlist = rinaNumpy.Divide_FloatArray_Float(ModeCharNumInfoChecksumBitlist, std);

        // 実行の確認用メッセージ
        Debug.Log($"{ReturnMyName()}が実行されました。");
        Debug.Log($"Version: {Version}, GridSize: {GridSize}");
        Debug.Log($"Normalized ModeCharNumInfoChecksumBitlist: {ModeCharNumInfoChecksumBitlist}");

        // プレイヤー自身を更新する処理
        return "Completed";
    }
}
